eureka/lib/manageevent.py

```diff
@@ -190,3 +192,113 @@ def updateevent(event, filename, add):
+    # First just check the specific inputdir folder
+    fnames = glob.glob(meta.inputdir+stage+'_'+meta.eventlabel+'*_Meta_Save.dat')
+    if len(fnames)==0:
+        # There were no metadata files in that folder, so let's see if there are in children folders
+        fnames = glob.glob(meta.inputdir+'**'+os.sep+stage+'_'+meta.eventlabel+'*_Meta_Save.dat', recursive=True)
+
+    if len(fnames)>=1:
+        # get the folder with the latest modified time
+        fname = max(fnames, key=os.path.getmtime)
+    
+    if len(fnames)==0 and allowFail:
+        # There may be no rateints files in the inputdir or any of its children directories - raise an error and give a helpful message
+        print('WARNING: Unable to find an output metadata file from Eureka!\'s '+stage
+             +' in the folder:\n"{}"\n'.format(meta.inputdir)
+             +'Assuming this '+stage+' data was produced by the JWST pipeline instead.')
+        return None, meta.inputdir, meta.inputdir_raw
+    elif len(fnames)==0:
+        # There may be no metafiles in the inputdir - raise an error and give a helpful message
+        raise AssertionError('Unable to find an output metadata file from Eureka!\'s '+stage
+                            +' in the folder:\n"{}"!'.format(meta.inputdir))
+    elif len(fnames)>1:
+        # There may be multiple runs - use the most recent but warn the user
+        print('WARNING: There are multiple metadata save files in the folder: \n"{}"\n'.format(meta.inputdir)
+                +'Using the metadata file: \n{}\n'.format(fname)
+                +'and will consider aperture ranges listed there. If this metadata file is not a part\n'
+                +'of the run you intended, please provide a more precise folder for the metadata file.')
+
+    fname = fname[:-4] # Strip off the .dat ending
+
+    # Load old savefile
+    old_meta = loadevent(fname)
+    # Code to not break backwards compatibility with old MetaClass save files but also use the new MetaClass going forwards
+    old_meta = readECF.MetaClass(**old_meta.__dict__)
+
+    old_meta.folder = os.sep.join(fname.split(os.sep)[:-1])+os.sep
+    old_meta.filename = fname.split(os.sep)[-1]
+
+    return old_meta, old_meta.folder, old_meta.folder[len(meta.topdir):]
+
+def mergeevents(new_meta, old_meta):
+    """Merge the current MetaClass data into the MetaClass object from a previous stage.
+
+    Parameters
+    ----------
+    new_meta : MetaClass
+        The metadata object for the current stage.
+    old_meta : MetaClass
+        The metadata object for the previous stage.
+
+    Returns
+    -------
+    new_meta : MetaClass
+        The current MetaClass object containing the details from the previous MetaClass object.
+
+    Notes
+    -------
+    History:
+
+    - April 25, 2022 Taylor Bell
+        Initial version.
+    """
+    # Load current ECF into old MetaClass
+    old_meta.read(new_meta.folder, new_meta.filename)
+
```

Comment:
Should we also check for conflicts between meta parameters to see if they have different values? The function could print a warning message if the same parameter has different values between meta instances. No need to throw an error. This is not high priority, just an idea.

Reply:
So the main reason I did things this way is to make sure that parameters from the new ECF overwrite the old ECF parameters (e.g. topdir, inputdir, outputdir, verbose). And for each aperture pair this function will get called again to overwrite the values to those needed for the new aperture (outputdir, spec_hw, bg_hw, and potentially inputdir). I can add log statements (perhaps with mute=True) if you want, but I personally don't think they'd be worthwhile given one of the main uses is specifically to overwrite parameters in the old meta object

Reply:
Ok, let's leave as is.